tests/post/steps/test_liveness.py

```diff
@@ -34,3 +45,43 @@ def check_exec(host, command, pod, namespace):
 
     with host.sudo():
         host.check_output(cmd)
+
+
+@then(
+    parsers.parse(
+        "we have {expected_nb_pods:d} running pod labeled '{label}'"
+    )
+)
+def count_running_pods(host, expected_nb_pods, label):
+    cmd = (
+        'kubectl --kubeconfig=/etc/kubernetes/admin.conf '
+        'get pod -l "{label}" --field-selector=status.phase=Running '
+        '--namespace kube-system -o json'
+    ).format(label=label)
+
+    # Just after the deployment, the pods may not in running state yet
+    # so implement a timeout
+    interval = 3
+    attempts = 10
+
+    with host.sudo():
+        for _ in range(attempts):
```

Comment:
shouldnt this be easier with a `while True` ?

Reply:
I don't know. I think it just a matter of taste. I prefer the `for` loop because you don't have to manipulate the `attempts` variable. With a `while True`, we'll have to decrease manually the value of `attempts` in the loop and then compare it to 0 to know if we reach the end of attempts or not